### PR TITLE
Add strict_allow_templates option for the dynamic mapping parameter

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

Add `strict_allow_templates` option for the dynamic mapping parameter, and remove the unsupported option `runtime`.

### Issues Resolved
https://github.com/opensearch-project/opensearch-api-specification/issues/407

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaobinlong** [comment]
The code PR of `strict_allow_templates` will be released in 2.16.0, so do we need to do some version check in this repo?

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added `concurrent_query_*` and `search_idle_reactivate_count_total` fields to `SearchStats` ([#395](https://github.com/opensearch-project/opensearch-api-specification/pull/395))
 - Added `remote_store` to `TranslogStats` ([#395](https://github.com/opensearch-project/opensearch-api-specification/pull/395))
 - Added `file` to `/_cache/clear` and `/{index}/_cache/clear` ([#396](https://github.com/opensearch-project/opensearch-api-specification/pull/396))
+- Add strict_allow_templates option for the dynamic mapping parameter ([#408](https://github.com/opensearch-project/opensearch-api-specification/pull/408))
```
Comment (dblock) [inline CHANGELOG.md:41]: Quote `strict_allow_templates`.

**Review by dblock — CHANGES_REQUESTED** [review]
Please add a test into tests. You should be able to write one for the `runtime` change, but for the new parameter you'll need to run against a newer version of OpenSearch locally.

**Comment by dblock** [comment]
> The code PR of `strict_allow_templates` will be released in 2.16.0, so do we need to do some version check in this repo?

Yes, this is https://github.com/opensearch-project/opensearch-api-specification/issues/398.

**Comment by gaobinlong** [comment]
The failed test seems related to the env, the test case works well locally.
image

**Comment by dblock** [comment]
> The failed test seems related to the env, the test case works well locally. image

I think 2.16 daily isn't starting, I'll look.

**Comment by dblock** [comment]
Update CI to a newer 2.16 build that has your feature and it should all go 🟢 ?

https://github.com/opensearch-project/opensearch-api-specification/blob/main/.github/workflows/test-spec.yml#L29

**Comment by gaobinlong** [comment]
> Update CI to a newer 2.16 build that has your feature and it should all go 🟢 ?
> 
> https://github.com/opensearch-project/opensearch-api-specification/blob/main/.github/workflows/test-spec.yml#L29

I think so.

**Comment by dblock** [comment]
> > Update CI to a newer 2.16 build that has your feature and it should all go 🟢 ?
> > https://github.com/opensearch-project/opensearch-api-specification/blob/main/.github/workflows/test-spec.yml#L29
> 
> I think so.

I mean you can do it in this PR and we can merge.

**Comment by gaobinlong** [comment]
@dblock I've updated the image version, now all checks have passed yet.

**Comment by dblock** [comment]
If you want to do more, maybe move the template tests into a subfolder, so `tests/indices/mapping/dynamic.yaml` and add some more tests for other mapping options?

**Comment by gaobinlong** [comment]
> If you want to do more, maybe move the template tests into a subfolder, so `tests/indices/mapping/dynamic.yaml` and add some more tests for other mapping options?

Sure, I'll do that later.

**Pull request merged**